src/Components/SuperAdmins/Form/index.jsx

```diff
@@ -35,14 +104,23 @@ const Form = (props) => {
 
   const onSubmit = (e) => {
     e.preventDefault();
-    isPut ? props.updateItem(props.itemId, item) : props.addItem(item);
+    if (isEdit) {
+      updateSuperAdmin(id, item);
+    } else {
+      addSuperAdmin(item);
+    }
+    history.push('/super-admins');
```

Comment:
You could move this push to the updateSuperAdmin and addSuperAdmin when the response is ok, in this way when I can't edit because is an error I stay in the form route.

Reply:
I agree with this idea :)

Reply:
fixed! thanks for the advice mates :)